devito/finite_differences/derivative.py

```diff
@@ -222,25 +220,37 @@ def _process_weights(cls, **kwargs):
 
     def __call__(self, x0=None, fd_order=None, side=None, method=None, weights=None):
         side = side or self._side
+        method = method or self._method
+        weights = weights if weights is not None else self._weights
+
         x0 = self._process_x0(self.dims, x0=x0)
         _x0 = frozendict({**self.x0, **x0})
```

Comment:
do we actually need the underscore for all these variable names ? 

it'd get way less verbose and then easier to read without the initial underscore

Reply:
For some yes but will try to cleanup a bit